### PR TITLE
Add actions for release workflow

### PR DESCRIPTION
**Pull request by hsaunders1904**
## Linked Issues

Partially addresses #1087 

## Description

This PR adds GitHub Actions workflows to produce releases of bluemira.

Please have a read through of the accompanying documentation to familiarise yourself with the process/suggest improvements.

### Release Notes

There is still the question of release notes. At the moment, these are auto-generated by the GitHub CLI, which basically lists out, and links to, pull requests that have been merged since the last release. This requires no extra work from us, but doesn't really offer users any indication to which changes are important to them.

Another option is [`towncrier'](https://github.com/twisted/towncrier). This will generate release notes for us, from small snippets that we push with every pull request. We create a file for each pull request, where we let it know what kind of pull request it is (a new feature, bugfix, documentation, etc.), an associated issue number, and write a small (often one line) comment on what's been added. When we need it to, it will build the snippets into a release notes document, and remove the snippets for that release. 

`towncrier` also has a `check` command, which we can run on the CI to remind us to add a release notes snippet.

Let me know what you think.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by je-cook — CHANGES_REQUESTED** [review]
Looks good, just a few minors.

On the changelog I think the no extra work option is fine for now. 
We may want to start highlighting changes more explicitly when we have full releases. 
Possibly we can comment in the release notes rst about that change to come in future

File: .github/workflows/release.yml
Comment (hsaunders1904) [reply .github/workflows/release.yml:8]: I believe you have to [have write access](https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event) to the repo to trigger the workflow. But I could look into further restrictions if need be.

**Comment by hsaunders1904** [comment]
Before merging, we should probably pick an initial release date. The scheduling uses the first release date to decide when a new release is due, so we need to hard-code that date in.

**Comment by je-cook** [comment]
> Before merging, we should probably pick an initial release date. The scheduling uses the first release date to decide when a new release is due, so we need to hard-code that date in.

Ok, I'll merge tomorrow after the meeting then and we can make that decision

**Comment by hsaunders1904** [comment]
@je-cook I've updated the release date, so this is ready to go in

**Pull request merged**
